File: Introibo/Data/Links/RelatedLink.swift
import Foundation

// MARK: - Related-link model
//
// A label + link-target string that content models carry as an optional
// `related` array. The target is a raw string parsed via LinkTarget.parse
// at read time (Phase 2 rendering will resolve it).
//
// Mirror of:
//   android/app/src/main/java/com/lampstandhq/introibo/data/links/RelatedLink.kt

struct RelatedLink: Decodable, Hashable {
    let label: String
    let target: String  // link-target string, parsed via LinkTarget.parse
}

File: android/app/src/main/java/com/lampstandhq/introibo/data/links/RelatedLink.kt
package com.lampstandhq.introibo.data.links

import kotlinx.serialization.Serializable

// MARK: - Related-link model
//
// A label + link-target string that content models carry as an optional
// `related` list. The target is a raw string parsed via LinkTarget.parse
// at read time (Phase 2 rendering will resolve it).
//
// Mirror of:
//   Introibo/Data/Links/RelatedLink.swift

@Serializable
data class RelatedLink(
    val label: String,
    val target: String,  // link-target string, parsed via LinkTarget.parse
)
